GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

```diff
@@ -0,0 +1,192 @@
+## Google Summer of Code - Application Template
+
+### Student
+
+**Name:** Chaitanya Prakash Bapat, Chaitanya
+
+**Affiliation:** Sardar Patel Institute of Technology (University of Mumbai), Computer Engineering, Bachelor of Engineering (BE)
+
+**Location:** Mumbai, India and India Time Zone UTC+05:30
+
+**GitHub handle:** [@ChaiBapchya](https://github.com/ChaiBapchya)
+
+**Slack handle:** @chaitanyabapat
+
+**Email:** chai.bapat@gmail.com
+
+**Other Contact:**  
+
+**Twitter** : [@ChaiBapchya](https://twitter.com/ChaiBapchya)
+
+**LinkedIn** : [/in/chaitanyaprakashbapat](https://www.linkedin.com/in/chaitanyaprakashbapat)
+
+### Project Details
+
+**Project Title:**  
+
+ML+NLP based Automated Change Monitoring
+
+**Project Abstract/Summary:**  
+
+Generating Environmental corpora, classifying change as relevant or not and prioritizing relevant ones using NLP.
+
+**Describe the need your project fills:**  
+
+_Broader Goal_ :- Categorizing and Prioritizing Changes / Diffs
+
+The project will fulfill the following needs - 
+
+1. _Curated Dataset_
+2. _Classifier_
+3. _Priority algorithm_
+
+**Describe how your project meets this need:**  
+
+Major setbacks
+1. Trump Budget announced - Savings of 31% EPA (reduction in fund allocation)
+2. Not a single corpus / dataset on environment can be found online
+3. Absence of an algorithm to track and prioritize changes
+
+Following needs will be met in my project :-
+1. Dataset - 
+
+After intense deliberation with analysts, subject matter experts (domain experts), we would stitch together a corpus of highly influential and important keywords in the domain of environment.
+
+2. Classifier - 
+
+Using ML techniques (classification algorithms), the diff (from PageFreezer,Versionista) will be categorized as to important, relevant or irrelevant ones by making use of the curated dataset.
+
+Output of Classifier will be cross-checked with the outcomes from NLP techniques (LDA and Summarizer)
+
+3. Priority Algorithm - 
+
+As currently, there is no standard way of prioritizing which change is to be given precendence over the others, which change is critical and more important than others - an algorithm would be needed for the same. Such an algorithm would be on the basis of "Confidence / Trust factor" and "Rating". Taking these 2 parameters into consideration, every change will be evaluated according to the algorithm and then prioritized.
+It would be a self-learning, ever-evolving process of curation.
```

Comment:
I have tried to explain - 
Sentence
1 - Need of Algo
2,3 - Parameters defining an algo